src/NzbDrone.Core/Validation/FileChmodValidator.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using FluentValidation;
+using Mono.Unix.Native;
+
+namespace NzbDrone.Core.Validation
+{
+    public static class FileChmodValidator
+    {
+        public static IRuleBuilderOptions<T, string> ValidFileChmod<T>(this IRuleBuilder<T, string> ruleBuilder)
+        {
+            return ruleBuilder.Must(x =>
+            {
+                try
+                {
+                    var permissions = NativeConvert.FromOctalPermissionString(x);
```

Comment:
I understand the conundrum, the only thing I can think of is to either add a IsValidFilePermissionMask function to DiskProvider or copy the flags and do it ourselves. First option is easier I think, DiskProviderBase would simply return throw new NotSupportedException(), with the real implementation in Sonarr.Mono. 
You'd need a non-static validator, similar to pathExistsValidator, to access IDiskProvider.

Reply:
Should be fixed now. :)